apps/trading/client-pages/markets/markets-page.tsx

```diff
@@ -13,58 +9,213 @@ import {
 } from '@vegaprotocol/environment';
 import { useT } from '../../lib/use-t';
 import { ErrorBoundary } from '../../components/error-boundary';
-import { MarketsSettings } from './markets-settings';
 import { usePageTitle } from '../../lib/hooks/use-page-title';
+import { Card } from '../../components/card';
+import { useDataProvider } from '@vegaprotocol/data-provider';
+import {
+  activeMarketsWithCandlesProvider,
+  calcCandleVolumePrice,
+  type MarketMaybeWithCandles,
+} from '@vegaprotocol/markets';
+import { useYesterday } from '@vegaprotocol/react-helpers';
+import { useEffect, useState } from 'react';
+import { Interval } from '@vegaprotocol/types';
+import { formatNumber } from '@vegaprotocol/utils';
+import { TopMarketList } from './top-market-list';
+import {
+  useNewListings,
+  useTVL,
+  useTopGainers,
+  useTotalVolume24hCandles,
+} from './use-markets-stats';
+import classNames from 'classnames';
+
+const POLLING_TIME = 2000;
 
 export const MarketsPage = () => {
+  const t = useT();
+  const yesterday = useYesterday();
+  const {
+    data: activeMarkets,
+    error,
+    reload,
+  } = useDataProvider({
+    dataProvider: activeMarketsWithCandlesProvider,
+    variables: {
+      since: new Date(yesterday).toISOString(),
+      interval: Interval.INTERVAL_I1H,
+    },
+  });
+  useEffect(() => {
+    const interval = setInterval(() => {
+      reload();
+    }, POLLING_TIME);
+    return () => {
+      clearInterval(interval);
+    };
+  }, [reload]);
+
+  usePageTitle(t('Markets'));
+
+  const topGainers = useTopGainers(activeMarkets);
+  const newListings = useNewListings(activeMarkets);
+  const totalVolume24hCandles = useTotalVolume24hCandles(activeMarkets);
+
+  const totalVolumeSparkline = (
+    <Sparkline width={80} height={20} data={totalVolume24hCandles} />
+  );
+  const tvl = useTVL(activeMarkets);
+
+  const totalVolume24h = activeMarkets?.reduce((acc, market) => {
+    return (
+      acc +
+      Number(
+        calcCandleVolumePrice(
+          market.candles || [],
+          market.decimalPlaces,
+          market.positionDecimalPlaces
+        )
+      )
+    );
+  }, 0);
+
+  return (
+    <ErrorBoundary feature="rewards">
+      <TinyScroll className="p-2 max-h-full overflow-auto">
+        <div className="flex flex-col w-full gap-3 p-3 xxl:px-[5.5rem]">
+          <div className="grid auto-rows-min grid-cols-3 lg:grid-cols-5 xl:grid-cols-3 gap-3 w-full">
+            <div className="flex flex-col gap-2 col-span-full lg:col-span-1">
+              <Card key="24h-vol" className="grow">
+                <div className="flex items-center justify-center w-full">
```

Comment:
Seems like some unnecessary markup and classes here. This could be cleaner

- The first element inside Card is unnecessary
- w-full is not required on the next element (block elements are full width by default)

Reply:
simplified it in latest commit